### PR TITLE
new: Add `site_type` field to linode_region and linode_regions data sources

### PR DESCRIPTION
**Pull request by lgarber-akamai**
## 📝 Description

This change adds a new `site_type` field to the linode_region and linode_regions data sources.

## ✔️ How to Test

The following test steps assume you have pulled down this PR locally.

### Unit Testing

```
make unit-test
```

### Integration Testing

```
make int-test PKG_NAME=linode/region
make int-test PKG_NAME=linode/regions
```

### Manual Testing

1. In a Terraform provider development environment (e.g. dx-devenv), plan the following configuration:

```terraform
# ...

data "linode_region" "test" {
  id = "us-mia"
}

data "linode_regions" "test" {
  filter {
    name = "site_type"
    values = ["core"]
  }
}

output "region_site_type" {
  value = {
    (data.linode_region.test.id) = data.linode_region.test.site_type
  }
}

output "regions_site_type" {
  value = {
    for region in data.linode_regions.test.regions :
    region.id => region.site_type
  }
}
```

2. Ensure the output looks similar to the following:

```
region_site_type = {
  "us-mia" = "core"
}
regions_site_type = {
  "ap-northeast" = "core"
  "ap-south" = "core"
  "ap-southeast" = "core"
  "ap-west" = "core"
  "br-gru" = "core"
  # ...
}
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -121,3 +121,6 @@ require (
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 	sigs.k8s.io/yaml v1.3.0 // indirect
 )
+
+// TODO: Remove before merging
+replace github.com/linode/linodego => github.com/linode/linodego v1.31.1-0.20240322191259-d349cae35701
```
Comment (lgarber-akamai) [inline go.mod:126]: This is necessary because site_type has _not_ been released in linodego quite yet

Comment (zliang-akamai) [inline go.mod:126]: Need a linodego release?

Comment (lgarber-akamai) [reply go.mod:126]: Yup!

**Review by jriddle-linode — APPROVED** [review]
LGTM

File: go.mod (hunk 1 above)
Comment (lgarber-akamai) [reply go.mod]: This functionality was released in v1.32.0 🙂

**Review by zliang-akamai — APPROVED** [review]
Nice work!

**Pull request merged**